test/integration/apiserver/cel/validatingadmissionpolicy_test.go

```diff
@@ -0,0 +1,937 @@
+			}, framework.SharedEtcd())
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer server.TearDownFn()
+
+			config := server.ClientConfig
+
+			client, err := clientset.NewForConfig(config)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if _, err := client.CoreV1().ConfigMaps("default").Create(context.TODO(), testcase.configMap, metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Create(context.TODO(), testcase.policy, metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicyBindings().Create(context.TODO(), testcase.policyBinding, metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			// TODO: add retry logic instead
+			time.Sleep(time.Second)
+
+			_, err = client.CoreV1().Namespaces().Create(context.TODO(), testcase.namespace, metav1.CreateOptions{})
+			if err == nil && testcase.err == "" {
+				return
+			}
+
+			if err == nil && testcase.err != "" {
+				t.Logf("actual error: %v", err)
+				t.Logf("expected error: %v", testcase.err)
+				t.Fatal("got nil error but expected an error")
+			}
+
+			if err != nil && testcase.err == "" {
+				t.Logf("actual error: %v", err)
+				t.Logf("expected error: %v", testcase.err)
+				t.Fatal("got error but expected none")
+			}
+
+			if err.Error() != testcase.err {
+				t.Logf("actual validation error: %v", err)
+				t.Logf("expected validation error: %v", testcase.err)
+				t.Error("unexpected validation error")
+			}
+
+			reason := err.(apierrors.APIStatus).Status().Reason
+			if reason != testcase.failureReason {
+				t.Logf("actual error reason: %v", reason)
+				t.Logf("expected failure reason: %v", testcase.failureReason)
+				t.Error("unexpected error reason")
+			}
+		})
+	}
+}
+
+func TestNoResource(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.CELValidatingAdmission, true)()
+	server, err := apiservertesting.StartTestServer(t, nil, nil, framework.SharedEtcd())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.TearDownFn()
+
+	config := server.ClientConfig
+
+	client, err := clientset.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	policy := withParams(&admissionregistrationv1alpha1.ParamKind{
+		APIVersion: "v1",
+		Kind:       "ConfigMap",
+	},
+		withNamespaceMatch(withFailurePolicy(admissionregistrationv1alpha1.Fail, makePolicy("test-policy"))))
+	policy.Spec.Validations = []admissionregistrationv1alpha1.Validation{
+		{
+			Expression: "true",
+		},
+	}
+	if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Create(context.TODO(), policy, metav1.CreateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+
+	// TODO (DangerOnTheRanger): add and subsequently remove config map here (instead of policy being invalid from the start)
```

Comment:
Is this TODO still relevant?

Reply:
The test as it stands never makes the config map - implementing the TODO would temporarily put the policy binding in a valid state, and there might be separate codepaths that get tested if that happens; do we test reconciling with a deleted param object anywhere else?

I think if we have a similar reconciliation test elsewhere I'd be on board for removing the TODO for sure, I'm a little worried there could be untested code otherwise.

Reply:
(marking this as resolved since the test is removed for now)